### PR TITLE
Add check for empty translation strings.

### PR DESCRIPTION
**Pull request by metanerd**
#### Summary
This is a workaround for this weblate issue: https://github.com/WeblateOrg/weblate/issues/4076

So this checks for occurrences like this:

```
{
      ...
      "admin.file_upload.uploadFile": "",
      ...
},
```

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
This PR was tested on:

### CODE REVIEW

**Comment by metanerd** [comment]
@enahum you mean in mmjstool? and then add this to package.json?

**Comment by enahum** [comment]
Yes in package.json. I want to do something like `npm run i18n-check` instead of make

**Comment by metanerd** [comment]
@enahum would this be enough, or do you want a regex implemented in javascript, so purely javascript?
```
# in package.json
    "i18n-checks": "make i18n-checks",
```

**Comment by enahum** [comment]
@metanerd basically I want to get rid of the Makefile in the near future ;) so I guess that would work for now but I rather not use make in any way

**Comment by metanerd** [comment]
OK, so I implement it in mmjstool or I put it somewhere in the mobile project, where would you prefer it?

**Comment by enahum** [comment]
0/5 maybe a script under the scripts folder, but if done in mmjstool it can be used in the webapp as well by just adding the script to the package.json

**Comment by metanerd** [comment]
OK, I will go with the mmjstool approach then.

File: .circleci/config.yml [hunk 1]
```diff
@@ -225,7 +225,7 @@ jobs:
           command: npm test
       - run:
           name: Check i18n
-          command: make i18n-extract-ci
+          command: make i18n-checks
```
Comment (enahum) [inline .circleci/config.yml:228]: can we use the npm script instead?

File: Makefile [hunk 2]
```diff
@@ -212,7 +212,7 @@ can-build-pr:
 i18n-extract: ## Extract strings for translation from the source code
 	npm run mmjstool -- i18n extract-mobile
 
-i18n-extract-ci:
+i18n-checks:
```
Comment (enahum) [inline Makefile:215]: I think we need both independently, one for checks the other one for extraction.. I can see them used  in two different use cases

File: package.json [hunk 3]
```diff
@@ -130,7 +130,7 @@
     "jetifier": "1.6.6",
     "jsdom-global": "3.0.2",
     "metro-react-native-babel-preset": "0.63.0",
-    "mmjstool": "github:mattermost/mattermost-utilities#086f4ffdca4e31a0be22f6bcdfa093ed83fb29e",
+    "mmjstool": "github:mattermost/mattermost-utilities#519b99a4e51e6c67a0dbd46a6efdff27dc835aaa",
```
Comment (enahum) [inline package.json:133]: missing the npm scripts

**Review by jespino — APPROVED** [review]

File: scripts/pre-commit.sh [hunk 4]
```diff
@@ -11,7 +11,7 @@ if [ -n "$jsfiles" ]; then
     for js in $jsfiles; do
         echo "$js"
         e=$(node_modules/.bin/eslint --quiet --fix $js)
-        if [[ -n "$e" ]]; then
+        if [ -n "$e" ]; then
```
Comment (enahum) [inline scripts/pre-commit.sh:14]: wondering if we could also add the check & the extract on the pre-commit ;) that could avoid a failed PR

File: Makefile [hunk 5]
```diff
@@ -197,7 +197,7 @@ unsigned-android: stop pre-build check-style prepare-android-build ## Build an u
 test: | pre-run check-style ## Runs tests
 	@npm test
 
-build-pr: | can-build-pr stop pre-build check-style i18n-extract-ci ## Build a PR from the mattermost-mobile repo
+build-pr: | can-build-pr stop pre-build check-style i18n-checks ## Build a PR from the mattermost-mobile repo
```
Comment (enahum) [inline Makefile:200]: At this point with  al the builds I'm inclined to remove all checks, styles, i18n, tests, etc.. cause the fastlane script actually checks if the git branch is dirty or not and I would expect that everything merged has been checked & tested.

thoughts?

File: scripts/pre-commit.sh (hunk 4 above)
Comment (metanerd) [reply scripts/pre-commit.sh:14]: True, added.

File: Makefile (hunk 5 above)
Comment (metanerd) [reply Makefile]: In general, I agree, if the fastlane script already does the checks. I think it is outside the scope of this PR though.

**Pull request merged**